packages/combo-box/src/vaadin-combo-box-light.js

```diff
@@ -206,6 +206,29 @@ class ComboBoxLight extends ComboBoxDataProviderMixin(ComboBoxMixin(ValidateMixi
       this._onClearAction();
     }
   }
+
+  /**
+   * @protected
+   * @override
+   */
+  _onFocusout(event) {
+    // prevent closing the overlay when moving focus from clear or toggle buttons to the internal input
+    if (this._isBlurringControlButtons(event) && this._isFocusingInputElement(event)) {
+      return;
+    }
+
+    super._onFocusout(event);
+  }
+
+  /** @protected */
+  _isFocusingInputElement(event) {
```

Comment:
I'd suggest inlining these helper methods directly into `_onFocusout` since it's unlikely that someone will override them.

```js
const isFocusingInputElement = ...
const isBlurringControlButtons = ...
```